.github/workflows/dbp_pr_checks.yml

```
@@ -0,0 +1,232 @@
+      - name: Run tests
+        run: set -o pipefail && swift test | tee -a build-log.txt | xcbeautify --report junit --report-path . --junit-report-filename package-tests.xml
+
+      - name: Publish Unit Tests Report
+        uses: mikepenz/action-junit-report@v4
+        if: always()
+        with:
+          check_name: Test Report (macOS)
+          report_paths: SharedPackages/DataBrokerProtectionShared/package-tests.xml
+          require_tests: true
+          check_retries: true
+
+      - name: Update Asana with failed unit tests
+        if: always() # always run even if the previous step fails
+        env:
+          ASANA_ACCESS_TOKEN: ${{ secrets.ASANA_ACCESS_TOKEN }}
+          WORKFLOW_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}
+        run: |
+          # Extract failed tests from the junit report
+          # Only keep failures unique by classname and name (column 1 and 2 of the yq output)
+          yq < tests.xml -p xml -o json -r \
+            $'[.testsuites.testsuite[].testcase] | flatten | map(select(.failure) | .+@classname + " " + .+@name + " \'" + .failure.+@message + "\' ${{ env.WORKFLOW_URL }}") | .[]' \
+            | sort -u -k 1,2 \
+            | xargs -L 1 ./scripts/report-failed-unit-test.sh -s ${{ vars.APPLE_CI_FAILING_TESTS_BSK_FAILED_TESTS_SECTION_ID }}
+
+      - name: Upload logs
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: build-log.txt
+          path: SharedPackages/DataBrokerProtectionShared/build-log.txt
+
+      - name: Fetch latest commit author
+        if: always() && github.ref_name == 'main'
+        id: fetch_commit_author
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          head_commit=$(git rev-parse HEAD)
+          author=$(gh api https://api.github.com/repos/${{ github.repository }}/commits/${head_commit} --jq .author.login)
+          echo "commit_author=${author}" >> $GITHUB_OUTPUT
+
+  unit-tests-ios:
+
+      name: Run unit tests (iOS)
+
+      runs-on: macos-15
+      timeout-minutes: 30
+
+      steps:
+
+      - name: Check out the code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set cache key hash
+        run: |
+          has_only_tags=$(jq '[ .pins[].state | has("version") ] | all' Package.resolved)
+          if [[ "$has_only_tags" == "true" ]]; then
+            echo "cache_key_hash=${{ hashFiles('SharedPackages/DataBrokerProtectionShared/Package.resolved') }}" >> $GITHUB_ENV
+          else
+            echo "Package.resolved contains dependencies specified by branch or commit, skipping cache."
+          fi
+
+      - name: Cache SPM
+        if: env.cache_key_hash
+        uses: actions/cache@v4
+        with:
+          path: SharedPackages/DataBrokerProtectionShared/DerivedData/SourcePackages
+          key: ${{ runner.os }}-bsk-spm-ios-${{ env.cache_key_hash }}
```

Same cache key comment here:

```suggestion
          key: ${{ runner.os }}-dbp-spm-ios-${{ env.cache_key_hash }}
```